### PR TITLE
Feature/220 refactor project structure

### PR DESCRIPTION
**Pull request by Shi-Ho**
# Overview

- Moved all code to a `/src` directory
- Merged APIs together using subAPI
- Updated APIs architecture to allow easy changes to paths prefix (ex: `/api`)
- Updated Dockerfile and docker-compose consequently
- Implemented Angular frontend -> URL is now http://localhost:4200
- Light documentation update to fit the changes

# Testing

Done with Swagger and angular front:
- autocomplete
- exact match
- fuzzy match

### To test before merging:
- **semantic match**
- **rag**

# Todo afterward

- Rethink documentation structure
- queries.py files implementation are a bit ugly, but might be removed once ORM implemented
- more refactoring on the code structure
- Update api paths

### CODE REVIEW

File: src/.env.example
Comment (Shi-Ho) [inline src/.env.example:1]: test @K-Schubert

File: src/docker-compose.yml
[the remark this answers is not in the document]
Comment (Shi-Ho) [reply src/docker-compose.yml:1]: done and tested

File: src/.env.example
[the remark this answers is not in the document]
Comment (Shi-Ho) [reply src/.env.example:1]: done and tested

File: src/copilot/app/rag/rag_processor.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+            Question and context documents language
+        k : int, default 0
+            Number of context documents to return (need to be revised, current logic in the code is confusing)
+        """
+        k = self.k_retrieve if k is None else k
+
+        rows = await semantic_similarity_match(request.query, db_name='embeddings', language=language, k=k)
+        documents = [dict(row) for row in rows][0]
+
+        return {"contextDocs": documents["text"], "sourceUrl": documents["url"], "cosineSimilarity": documents["distance"]}
```
Comment (K-Schubert) [inline src/copilot/app/rag/rag_processor.py:57]: change 

- return {"contextDocs": documents["text"], "sourceUrl": documents["url"], "cosineSimilarity": documents["distance"]} 

to

- return {"contextDocs": documents["text"], "sourceUrl": documents["url"], "similarity_metric": documents["similarity_metric"]}

File: src/copilot/app/rag_api.py [hunk 2]
```diff
@@ -30,23 +39,26 @@
 )
 
 
-@app.post("/rag/process", summary="Process RAG query endpoint", response_description="Return result from processing RAG query", status_code=200)
+@app.post("/query",
```
Comment (K-Schubert) [inline src/copilot/app/rag_api.py:42]: rename api route to "/process" as copilot-frontend currently makes calls to "/rag/process" route.

File: docker-compose.yml
Comment (K-Schubert) [inline docker-compose.yml:1]: Had to change:

volumes:
    - ./pgadmin/servers.json:/pgadmin4/servers.json

to

volumes:
    - ./src/pgadmin/servers.json:/pgadmin4/servers.json

Can you run the docker-compose without this change?

File: src/copilot/app/rag_api.py [hunk 3]
```diff
@@ -30,23 +39,26 @@
...
+          summary="Process RAG query endpoint",
+          response_description="Return result from processing RAG query",
+          status_code=200)
 async def process_query(request: RAGRequest):
     content = await processor.process(request)
     return StreamingResponse(content, media_type="text/event-stream")
 
 
-@app.post("/rag/docs", summary="Retrieve context docs endpoint", response_description="Return context docs from semantic search", status_code=200)
+@app.post("/context",
```
Comment (K-Schubert) [inline src/copilot/app/rag_api.py:51]: I would stick with "/docs" as the route for retrieving docs although we don't use the endpoint at the moment. It's considered the standard nomenclature (to be more precise "context_docs", but I think /docs is sufficient).

File: src/copilot/app/rag_api.py (hunk 2 above)
Comment (Shi-Ho) [reply src/copilot/app/rag_api.py:42]: I changed in the route call in the frontend to match /query

File: src/copilot/app/rag_api.py (hunk 3 above)
Comment (Shi-Ho) [reply src/copilot/app/rag_api.py]: I changed to /context_docs. I want to avoid confusion with the swagger doc

**Pull request merged**